src/sbd_helper.in

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# Authors:
+#  Klaus Wenninger <kwenning@redhat.com>
+#
+# License: Revised BSD
+
+# chkconfig: - 99 01
+# description: Shared-storage based fencing daemon
+# processname: sbd
+#
+### BEGIN INIT INFO
+# Provides:		sbd_helper
+# Required-Start:	$local_fs $remote_fs
+# Should-Start:		$syslog
+# Required-Stop:	$local_fs $remote_fs
+# Default-Start:
+# Default-Stop:
+# Short-Description:	Starts and stops Shared-storage based fencing daemon.
+# Description:		Starts and stops Shared-storage based fencing daemon.
+### END INIT INFO
+
+desc="Shared-storage based fencing daemon"
+prog="sbd"
+pidfile="/var/run/${prog}.pid"
+
+# set secure PATH
+PATH="/sbin:/bin:/usr/sbin:/usr/bin:/usr/sbin"
+
+checkrc() {
+	if [ $? = 0 ]; then
+		success
+	else
+		failure
+	fi
+}
+
+success()
+{
+	echo -ne "[  OK  ]\r"
+}
+
+failure()
+{
+	echo -ne "[FAILED]\r"
+}
+
+log()
+{
+	logger -t sbd -p daemon.notice "$*"
+}
+
+notify()
+{
+	log "$*"
+	echo -n "   $*"
+}
+
+sbd_status()
+{
+	pid=$(cat $pidfile 2>/dev/null)
+	rtrn=$?
+	if [ $rtrn -eq 0 ]; then
+		kill -0 $pid > /dev/null 2>&1
+		rtrn=$?
+	fi
+	if [ $rtrn -ne 0 ]; then
+		rm -f $pidfile
+		echo "$prog is stopped"
+		rtrn=3
+	else
+		echo "$prog (pid $pid) is running..."
+	fi
+	return $rtrn
+}
+
+# rpm based distros
+if [ -d /etc/sysconfig ]; then
+	[ -f /etc/rc.d/init.d/functions ] && . /etc/rc.d/init.d/functions
+set -a
+	[ -f /etc/sysconfig/sbd ] && . /etc/sysconfig/sbd
+set +a
+	[ -z "$LOCK_FILE" ] && LOCK_FILE="/var/lock/subsys/sbd"
+fi
+
+# deb based distros
+if [ -d /etc/default ]; then
+set -a
+[ -f /etc/default/sbd ] && . /etc/default/sbd
+set +a
+[ -z "$LOCK_FILE" ] && LOCK_FILE="/var/lock/sbd"
+fi
```

Comment:
Pacemaker now uses a configure option to set @CONFIGDIR@ to simplify the above section. Something like that might be worthwhile here (see pacemaker commits 5f234920 and c2684cc8).